pkg/source/adapter/adapter_test.go

```diff
@@ -441,9 +440,24 @@ func TestAdapter_Start(t *testing.T) { // just increase code coverage
 	_ = os.Setenv("KAFKA_BOOTSTRAP_SERVERS", "my-cluster-kafka-bootstrap.my-kafka-namespace:9092")
 
 	a := NewAdapter(ctx, NewEnvConfig(), nil, nil)
-	require.Panics(t, func() {
-		_ = a.Start(ctx)
-	})
+
+	stopped := make(chan bool, 1)
+	go func() {
+		err := a.Start(ctx)
+		if err != nil {
+			t.Error("unexpected error:", err)
```

Comment:
What about adding a status field? marking whether or not the consumer group was successfully started or not?

Reply:
how do you propagate such status back to the controller/crd?

Reply:
we can leverage @devguyio work on channel subscriptions for that. The controller does it, not the dispatcher/adapter.

When we get this, this PR becomes irrelevant, so closing